File: app/src/main/java/com/pedrosoares/cotacaoapp/presentation/CoinsContract.java
package com.pedrosoares.cotacaoapp.presentation;

import com.pedrosoares.cotacaoapp.core.base.BaseContract;
import com.pedrosoares.cotacaoapp.model.domain.CoinsDomain;

public interface CoinsContract {

    interface CoinsListView {
        void populateCoins(CoinsDomain coinsDomain);
        void success();
        void loading();
        void error();
    }

    interface CoinsListPresenter extends BaseContract.Presenter {
        void fetchCoins();
    }
}

File: app/src/main/java/com/pedrosoares/cotacaoapp/presentation/CoinsContract.kt
package com.pedrosoares.cotacaoapp.presentation

import com.pedrosoares.cotacaoapp.core.base.BaseContract.Presenter
import com.pedrosoares.cotacaoapp.model.domain.CoinsDomain

interface CoinsContract {
    interface CoinsListView {
        fun populateCoins(coinsDomain: CoinsDomain)
        fun success()
        fun loading()
        fun error()
    }

    interface CoinsListPresenter : Presenter {
        fun fetchCoins()
    }

    interface ListenerLayout {
        fun verifyLayout(): Boolean
    }
}